File: data/multilang_comparison/program_cpp_array/src/func33.cpp
#include "program_cpp_array.hpp" 
Array* func33(Array_param* vars, int loopsFactor) {
   size_t pCounter = vars->size;
   Array* array85;
   if (pCounter > 0) {
      array85 = vars->data[--pCounter];
      array85->refC++;
      DEBUG_COPY(array85->id);
   } else {
      array85 = new Array();
      array85->size = 163;
      array85->refC = 1;
      array85->id = 85;
      array85->data = new unsigned int[array85->size]();
      DEBUG_NEW(array85->id);
   }
   Array_param params0;
   params0.size = 1;
   Array* data_params0[1];
   params0.data = data_params0;
   params0.data[0] = array85;
   Array* array86 = func36(&params0, get_path(), loopsFactor);
   DEBUG_RETURN(array86->id);
   if(params0.size > 0) {
   	 params0.data.clear();
   }
   unsigned int loop26 = 0;
   unsigned int loopLimit26 = (50)/3 + 1;
   for(; loop26 < loopLimit26; loop26++) {
      for (int i = 0; i < array86->size; i++) {
         array86->data[i]++;
      }
   }
   for (int i = 0; i < array85->size; i++) {
       if (array85->data[i] == 59) {
           return array85;
       }
   }
   array85->refC--;
   if(array85->refC == 0) {
      delete[] array85->data;
      array85->data = nullptr;
      DEBUG_FREE(array85->id);
      delete array85;
   }
   return array86;
}

File: data/multilang_comparison/program_go/src/func33.go
package main
import "math/rand"
var _ = rand.Intn
func Func33(vars *ArrayParam, loopsFactor int) *Array{
   var pCounter int = vars.Size;
   _ = pCounter
   var array85 *Array
   if pCounter > 0 {
       array85 = vars.Data[pCounter-1]
       pCounter--
       array85.RefC++
   } else {
       array85 = &Array{
           Size: 163,
           RefC: 1,
           Id: 85,
           Data: make([]uint32, 163),
       }
   }
   var params0 ArrayParam
   params0.Size = 1
   params0.Data = make([]*Array, params0.Size)
   params0.Data[0] = array85
   var array86 *Array = Func36(&params0, getPath(), loopsFactor);
   params0.Data = nil
   params0.Size = 0
   var loop26 int = 0;
   var loopLimit26 int = (50)/3 + 1;
   for ; loop26 < loopLimit26; loop26++ {
      for i := 0; i < array86.Size; i++ {
          array86.Data[i]++
      }
   }
   for i := 0; i < array85.Size; i++ {
       if array85.Data[i] == 59 {
           return array85
       }
   }
   array85.RefC--
   if array85.RefC == 0 {
       array85.Data = nil
       array85 = nil
   }
   return array86;
}

